src/home/celebration/NftReward.tsx

```diff
@@ -0,0 +1,166 @@
+import { useSafeAreaInsets } from 'react-native-safe-area-context'
+import { HomeEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { openDeepLink } from 'src/app/actions'
+import { BottomSheetRefType } from 'src/components/BottomSheet'
+import BottomSheetBase from 'src/components/BottomSheetBase'
+import Button, { BtnSizes, BtnTypes } from 'src/components/Button'
+import { nftRewardDisplayed } from 'src/home/actions'
+import { isSameNftContract } from 'src/home/celebration/utils'
+import { nftCelebrationSelector, showNftRewardSelector } from 'src/home/selectors'
+import i18n from 'src/i18n'
+import { nftsWithMetadataSelector } from 'src/nfts/selectors'
+import { useDispatch, useSelector } from 'src/redux/hooks'
+import Colors from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import { formatDistanceToNow } from 'src/utils/time'
+
+export default function NftRewardBottomSheet() {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+
+  const canShowNftReward = useSelector(showNftRewardSelector)
+  const nftCelebration = useSelector(nftCelebrationSelector)
+
+  const nfts = useSelector(nftsWithMetadataSelector)
+  const matchingNft = useMemo(
+    () => nfts.find((nft) => isSameNftContract(nft, nftCelebration)),
+    [nftCelebration]
+  )
+
+  const isVisible = canShowNftReward && matchingNft
+
+  const insets = useSafeAreaInsets()
+  const insetsStyle = { paddingBottom: Math.max(insets.bottom, Spacing.Regular16) }
+
+  const bottomSheetRef = useRef<BottomSheetRefType>(null)
+
+  const expirationDate = new Date(nftCelebration?.expirationDate ?? 0)
+  const rewardReminderDate = new Date(nftCelebration?.rewardReminderDate ?? 0)
+
+  const aboutToExpire = isToday(rewardReminderDate) || isPast(rewardReminderDate)
+  const expirationLabelText = formatDistanceToNow(expirationDate, i18n, { addSuffix: true })
+
+  const { pillStyle, labelStyle } = aboutToExpire
+    ? {
+        pillStyle: { backgroundColor: Colors.warningLight },
+        labelStyle: { color: Colors.warningDark },
+      }
+    : {
+        pillStyle: { backgroundColor: Colors.gray1 },
+        labelStyle: { color: Colors.black },
+      }
+
+  const copyText = aboutToExpire ? 'rewardReminderBottomSheet' : 'rewardBottomSheet'
+
+  useEffect(() => {
+    if (isVisible) {
+      // Wait for the home screen to have less ongoing async tasks.
+      // This should help the bottom sheet animation run smoothly.
+      const timeoutId = setTimeout(() => {
+        bottomSheetRef.current?.expand()
+      }, 1000)
+      return () => clearTimeout(timeoutId)
+    }
+  }, [isVisible])
+
+  const handleBottomSheetPositionChange = (index: number) => {
+    if (!nftCelebration) {
+      return // This should never happen
+    }
+
+    if (index === -1) {
+      ValoraAnalytics.track(HomeEvents.nft_reward_dismiss, {
```

Comment:
i think this is fine, but i think this dismiss event will always be fired (even if the accept event is also fired) - perhaps something we should note in the docs so that it's clear how to use this event. currently the doc says that this event is only triggered when the user dismisses the sheet (which i interpreted as dismiss without accepting the reward)

Reply:
that's a great catch! thanks!
the docs are correct: this event must dispatch only if the reward wasn't accepted. fixed it by adding a state variable.

sadly I can't figure out yet how to simulate bottom sheet dismiss event in tests to cover it with a test.